monitoring/mongodb/alerts.yaml

```diff
@@ -139,3 +139,45 @@ groups:
     annotations:
       description: 'MongoDb has low disk space'
       summary: 'MongoDb has low disk space'
+  
+  - alert: MongoDbNodeRecovering
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"}[1h]) == 3
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "The Mongodb instance `{{ $labels.pod }}` is in the 'RECOVERING' state for over an hour. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time. This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed."
```

Comment:
> This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed.

is it really ?

Reply:
When we perform a full init sync, if the cluster is still processing operations, there is a risk that we never converge, if the recovering takes longer than the oplog history...
The recovering instance might be in this state for more than one hour on most of the deployed platforms today